ipatests/test_xmlrpc/test_group_plugin.py

```diff
@@ -466,6 +466,13 @@ def test_search_for_all_nonposix(self, group):
             ],
         ), result)
 
+    def test_upgrade_nonposix_to_posix_and_external(self, group):
+        """ Update non-posix group to promote it to posix group & external"""
+        command = group.make_update_command(dict(posix=True, external=True))
+        with raises_exact(errors.MutuallyExclusiveError(
+                reason=u"gid cannot be set for external group")):
+            command()
+
```

Comment:
You can also add a test that does `ipa group-mod --gid=12345 --external group` on a non-posix group. It should return the error message. Then `ipa group-mod --gid=12345` on the same non-posix group, it should update the group into a posix one.

Reply:
Sure. I think there is already a non-posix to posix group_mod test but I'll double check.